### ISSUE TITLE
Could not cast value of type 'JavaUtilArrays_ArrayList'  to 'JavaUtilArrayList'

### ISSUE THREAD

**Issue by confile** [google/j2objc#614]
I translated the following code: 

```
List<String> imageList = new ArrayList<>();
imageList = Arrays.asList(imageUrlString.split(","));    
dto.setImageUrls(imageList);
```

In swift I was doing: 

```
for url in dto.getImageUrls() as! JavaUtilArrayList {
   imageUrls.append(url as! String)
}
```

At runtime I get the error: 

```
Could not cast value of type 'JavaUtilArrays_ArrayList'  to 'JavaUtilArrayList'
```

**Comment by confile** [comment]
It only works this way: 

```
imageList.addAll(Arrays.asList(imageUrlString.split(","))); 
```

This is strange because ``Arrays.asList()`` returns a ``List<T>`` so there should be no problem.

**Comment by tomball** [comment]
Arrays.toList() is declared as returning a List instance, not an ArrayList, and Swift caught you. This should work instead, although I'm not a Swift expert (at least, the types should match):

```
for url in dto.getImageUrls() as! JavaUtilList {
   imageUrls.append(url as! String)
}
```

**Comment by confile** [comment]
thank you

**Comment by confile** [comment]
@tomball No this does not work since: 

> Type 'JavaUtilList' does not conform to protocol 'SequenceType'

**Comment by lukhnos** [comment]
It appears you can't enumerate a `JavaUtilList` in Swift – in fact you can't enumerate any `NSFastEnumeration` objects directly, [according to this](http://stackoverflow.com/questions/25788290/nsfastenumeration-in-swift). The workaround is to extend the concrete list class you're using. Assuming you get an `JavaUtilArrayList` from some function `foo()`, and assuming the elements are strings:

```swift
extension JavaUtilArrayList: SequenceType {
    public func generate() -> NSFastGenerator {
        return NSFastGenerator(self)
    }
}

var jual: JavaUtilArrayList = foo()       
for x in jual {
    NSLog(x as! String)
}
```

This technique doesn't apply to `JavaUtilList`. Although protocol extensions are available as of Swift 2.0, they cannot have an inheritance clause.

**Comment by confile** [comment]
You should it do more generally: 

```
extension JavaUtilAbstractCollection : SequenceType {

  public func generate() -> NSFastGenerator {
    return NSFastGenerator(self)
  }    
}
```
